src/rendergraph/common/rendergraph/nodeinterface.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "backend/basenode.h"
+
+namespace rendergraph {
+
+template<class T_Node>
+class NodeInterface : public T_Node {
+  public:
+    void appendChildNode(std::unique_ptr<BaseNode>&& pNode) {
+        T_Node::appendChildNode(pNode.release());
```

Comment:
see the exception for unique_ptr https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-consume

```suggestion
    void appendChildNode(std::unique_ptr<BaseNode> pNode) {
        T_Node::appendChildNode(pNode.release());
```

Reply:
Thanks for spotting this. I will change this here and elsewhere.

> I still highly appreciate that you're keeping this alive.

It has been very interesting work, and I like seeing it becoming increasingly more elegant! Also @acolombier moving forward with the QML counterpart has been very inspirational. It took a couple of iterations but we now have the same waveform renderer code showing the waveforms with both the OpenGL and the Qt Scenegraph backend, and smooth too. It all still needs a bit of polishing but we are getting close.

As for this PR, I think it will be best if I rebase and squash when I think I am done, and request a new review.

Reply:
if possible it be nice if you can retain some structure (instead of quashing all into one) so its a little easier to review. It'd be nice if the abstract interfaces were in a separate commit to the two implementations so forth if feasible.

Reply:
Yes, I will try to split it in multiple PRs, but a lot of the commits are not relevant anymore, so I will first squash and then create some new PRs with parts of the code.